### PR TITLE
Dedicated Options for Flags

### LINKED ISSUE #19: Setting Flags
I think it would make sense to replace the `flags` parameter with specific config keys.

## Sources Provision

1. `PersistentVolumeClaim` / `annotations`
   - formatted like `gcs.csi.ofek.dev/foo-bar: xyz`
   - forwarded to `PersistentVolume` / `volumeContext` on `CreateVolume`
1. `StorageClass` / `parameters`
   - formatted like `gcs.csi.ofek.dev/foo-bar: xyz`
   - forwarded to `PersistentVolume` / `volumeContext` on `CreateVolume`
1. `StorageClass` / `MountOptions`
   - formatted as string like `--foo-bar="xyz"`
1. `StorageClass` / `csi.storage.k8s.io/provisioner-secret`
   - formatted like `fooBar: xyz`
   - forwarded to `PersistentVolume` / `volumeContext` on `CreateVolume`

## Sources Node

1. `PersistentVolume` / `volumeContext`
   - formatted like `fooBar: xyz`
1. `PersistentVolume` / `MountOptions`
   - formatted as string like `--foo-bar="xyz"`
1. `PersistentVolume` / `nodePublishSecretRef`
   - formatted like `fooBar: xyz`

## Priority

In order of sources.

## Flags

* ~~`foreground`~~ - Not supported since this should be internal to the driver
* `o` - No Validation - Special Note about how this may break stuff, Linking [here](https://manpages.debian.org/testing/fuse/mount.fuse.8.en.html), Internally Renamed to `fuseMountOption`
* `dir-mode` - Validate Number (Octal, permission)
* `file-mode` - Validate Number (Octal, permission)
* `uid` - Validate Number
* `gid` - Validate Number
* `implicit-dirs` - Validate Boolean (`"true"` / `"false"`)
* ~~`only-dir`~~ - Not supported since there is a [K8S way](https://kubernetes.io/docs/concepts/storage/volumes/#using-subpath) for that
* `billing-project` - Validate String
* ~~`key-file`~~ - Not supported since this should be internal to the driver
* `limit-bytes-per-sec` Validate Number
* `limit-ops-per-sec` Validate Number
* `stat-cache-ttl` Validate Number
* `type-cache-ttl` Validate Number
* ~~`temp-dir`~~ - Not supported since this should be internal to the driver
* ~~`debug_*`~~ - Not supported since this should be internal to the driver
* ~~`help`~~ - Not supported since that makes no sense
* ~~`version`~~ - Not supported since that makes no sense

## Docs

* Document all location / flag names
* Add Doc Page / Section for running containers not as root (#6)
  - Always
    * Recommend setting `gid` to any number
    * Recommend setting `Pod/securityContext/fsGroup` to same value
    * Recommend setting `Pod/securityContext/fsGroupChangePolicy` to `OnRootMismatch` to avoid changes since our FS does not support permissions
  - All containers inside pod have same UID
    * Recommend setting `uid` to user id of container
  - Containers inside pod have different UID
    * Set `dir-mode` to `077x` (Allow all users of group)
    * Set `file-mode` to `066x` (Allow all users of group)

## Discussions

* Make setup for usage with non-root users easier?
  - `dir-mode` default `0775` (instead `0755`)
  - `file-mode` default `0664` (instead `0644`)
  - `gid` default `63147` (randomly selected number )
  - This would shorten "Non-Root-Container" Docs to only do the steps `Pod/securityContext/fsGroup` and `Pod/securityContext/fsGroupChangePolicy`

### PR DESCRIPTION
**Pull request by maennchen**
Closes #19

### CODE REVIEW

File: pkg/flags/flags.go [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+	if location != "" {
+		result[FLAG_LOCATION] = location
+	}
+
+	if len(fuseMountOptions) != 0 {
+		result[FLAG_FUSE_MOUNT_OPTION] = strings.Join(fuseMountOptions, " ")
+	}
+
+	if dirMode != -1 {
+		result[FLAG_DIR_MODE] = "0" + strconv.FormatInt(dirMode, 10)
```
Comment (maennchen) [inline pkg/flags/flags.go:287]: If we use base we 10, the 0 prefix is unnecessary.

Comment (maennchen) [reply pkg/flags/flags.go:287]: I'll research an option to custom parse it always octal.

Comment (maennchen) [reply pkg/flags/flags.go]: @ofek What do you think about this change?

File: pkg/flags/flags.go [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+	ANNOTATION_BILLING_PROJECT     = "gcs.csi.ofek.dev/billing-project"
+	ANNOTATION_LIMIT_BYTES_PER_SEC = "gcs.csi.ofek.dev/limit-bytes-per-sec"
+	ANNOTATION_LIMIT_OPS_PER_SEC   = "gcs.csi.ofek.dev/limit-ops-per-sec"
+	ANNOTATION_STAT_CACHE_TTL      = "gcs.csi.ofek.dev/stat-cache-ttl"
+	ANNOTATION_TYPE_CACHE_TTL      = "gcs.csi.ofek.dev/type-cache-ttl"
+
+	MOUNT_OPTION_BUCKET              = "bucket"
+	MOUNT_OPTION_PROJECT_ID          = "project-id"
+	MOUNT_OPTION_LOCATION            = "location"
+	MOUNT_OPTION_FUSE_MOUNT_OPTION   = "fuse-mount-option"
```
Comment (ofek) [inline pkg/flags/flags.go:47]:
```suggestion
	MOUNT_OPTION_FUSE_MOUNT_OPTION   = "fuse-mount-options"
```

**Pull request merged**